File: DouDizhuClient/Assets/Scripts/Gameplay/Player/Player.cs
using System.Threading.Tasks;
using Network.Proto;

namespace Gameplay.Player
{
    public class PlayerManager
    {
        public bool IsLogin => !string.IsNullOrEmpty(ID);
        public string ID { get; private set; }
        public string Name { get; private set; }

        private static PlayerManager s_Instance;
        public static PlayerManager Instance
        {
            get
            {
                if (s_Instance == null)
                    s_Instance = new PlayerManager();
                return s_Instance;
            }
        }

        public async Task<bool> Register(string account, string password)
        {
            var response = await Network.NetworkManager.Instance.RequestAsync(PMsgId.Register, new PRegisterRequest()
            {
                Account = account,
                Password = password
            });
            if (!response.IsSuccess)
                return false;
            return true;
        }
        
        public async Task<bool> Login(string account, string password)
        {
            var response = await Network.NetworkManager.Instance.RequestAsync<PLoginRequest, PLoginResponse>(PMsgId.Login, new PLoginRequest()
            {
                Account = account,
                Password = password
            });
            if (!response.IsSuccess)
                return false;

            ID = response.Data.PlayerId;
            return true;
        }
    }
}

File: DouDizhuServer/scripts/network/handler/player.go
package handler

import (
	"DouDizhuServer/scripts/errordef"
	"DouDizhuServer/scripts/gameplay/player"
	"DouDizhuServer/scripts/network/message"
	"DouDizhuServer/scripts/network/protodef"
	"DouDizhuServer/scripts/network/translator"

	"google.golang.org/protobuf/proto"
)

func HandleRegister(context *message.MessageContext, req *proto.Message) (*message.HandleResult, error) {
	reqMsg, ok := (*req).(*protodef.PRegisterRequest)
	if !ok {
		return nil, errordef.NewGameplayError(errordef.CodeInvalidRequest)
	}
	account := reqMsg.GetAccount()
	password := reqMsg.GetPassword()
	err := player.Manager.Register(account, password)
	if err != nil {
		return nil, err
	}
	return &message.HandleResult{}, nil
}

func HandleLogin(context *message.MessageContext, req *proto.Message) (*message.HandleResult, error) {
	reqMsg, ok := (*req).(*protodef.PLoginRequest)
	if !ok {
		return nil, errordef.NewGameplayError(errordef.CodeInvalidRequest)
	}
	account := reqMsg.GetAccount()
	password := reqMsg.GetPassword()

	player, err := player.Manager.Login(account, password, context.SessionId)
	if err != nil {
		return nil, err
	}

	result := &message.HandleResult{
		Resp: &protodef.PLoginResponse{
			Info: translator.PlayerToProto(player),
		},
	}
	return result, nil
}
